src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestException.cs

```diff
@@ -28,6 +28,23 @@ public HttpRequestException(string message, Exception inner)
             }
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="HttpRequestException" /> class with a specific message that describes the current exception, an inner exception, and an HTTP status code.
+        /// </summary>
+        /// <param name="message">A message that describes the current exception.</param>
+        /// <param name="inner">The inner exception.</param>
+        /// <param name="statusCode">The HTTP status code.</param>
+        public HttpRequestException(string message, Exception inner, HttpStatusCode? statusCode)
+            : this(message, inner)
+        {
+            StatusCode = statusCode;
+        }
+
+        /// <summary>
+        /// The HTTP status code to be returned with the exception.
```

Comment:
```suggestion
        /// Gets the HTTP status code associated with the error condition represented by the exception.
```

Reply:
I would also indicate that it might be null in the case than the error is not sourced from status code.

Reply:
Other docs I tried to learn from / imitate seem to treat nullable values as though they are not null, i.e. they do not tend to refer to it's nullability.

e.g. [ContentRangeHeaderValue.Length](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.headers.contentrangeheadervalue.length?view=netframework-4.8#System_Net_Http_Headers_ContentRangeHeaderValue_Length).

Reply:
@stephentoub what's the convention here? If I look at existing docs, such as [Exception.Message](https://docs.microsoft.com/en-us/dotnet/api/system.exception.message?view=netframework-4.8#System_Exception_Message) or the one linked above, the main page starts with "Gets the..." but then there's also a little "Property Value" section that seems to indicate what the property represents, rather than what the property's getter method does.

Which one of these is represented by the `<summary>` and how would I annotate the code to define the other one?

Reply:
@carlossanlop @mairaw do you have an opinions here

Reply:
I do think we will have inevitable issues filed asking why there is no status code, if we do not document it. Especially as this is a high-level feature.

Reply:
The summary should start with "Gets the...".

And lots of docs describe what null means (often also discussing Nothing in Visual Basic). It's fine to say something like ", or null if the exception was created for a reason other than a non-success status code."

Reply:
Documentation for properties should have a `<summary>` and a `<value>`. [Here's the xml in dotnet-api-docs](https://github.com/dotnet/dotnet-api-docs/blob/master/xml/System/Exception.xml#L1044) representing the `Exception.Message`.

I'm not sure why Visual Studio does not automatically add the `<value>`, but please add it if you can, @yaakov-h . And yes, the summary represents what the getter or setter does, while the value describes what the property represents.

If you're curious, [here is our documentation](https://github.com/dotnet/docs/blob/master/docs/csharp/properties.md) on how to document a property.

Reply:
Given that VS doesn't automatically add the value tag, I consider those optional. 

Our wiki has our rules on how to write a property summary: https://github.com/dotnet/dotnet-api-docs/wiki/Summary%3A-Property